### PR TITLE
docs(contributing): add commits section

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR adds some note to `CONTRIBUTING.md` around clean commits. 

## Checklist

- [x] updated `CHANGELOG.md`

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -23,9 +24,12 @@ you'd like to address unless the proposed fix is minor.
...
 In your Pull Requests (PR), link to the issue that the PR solves.
 
-Please ensure that the base of your PR is the **master** branch. (Note: The default
-GitHub branch is the latest release branch, though you should point all of your changes to be merged into
-master).
+Please ensure that the base of your PR is the **main** branch.
+
+### Commits
+
+We prefer a clean commit history. Please clean up your commits before marking your PR
```
Comment (jawnsy) [inline docs/CONTRIBUTING.md:31]: What does "clean up your commits" mean int his context? Does it mean that we should make sure to squash all commits before asking for review (cleanup, squash, force-push), or does it just mean to make sure that debug printf's etc are removed (possibly as an additional commit)?

Comment (jsjoeio) [reply docs/CONTRIBUTING.md:31]: Good question! It means we should squash all commits before asking for review. I'll add that sentence!

Comment (jsjoeio) [reply docs/CONTRIBUTING.md]: Updated!

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -23,9 +24,11 @@ you'd like to address unless the proposed fix is minor.
...
 In your Pull Requests (PR), link to the issue that the PR solves.
 
-Please ensure that the base of your PR is the **master** branch. (Note: The default
-GitHub branch is the latest release branch, though you should point all of your changes to be merged into
-master).
+Please ensure that the base of your PR is the **main** branch.
+
+### Commits
+
+We prefer a clean commit history. This means you should squash all commits before asking for review (cleanup, squash, force-push). If you need help with this, feel free to leave a comment in your PR and we'll guide you.
```
Comment (code-asher) [inline docs/CONTRIBUTING.md:31]: `squash all commits` == PRs should only have one commit? Or does that just mean to squash fixups and fixup-type commits?

Comment (jsjoeio) [reply docs/CONTRIBUTING.md:31]: Squash fixups and fixup-type commits - let me reword to clarify! Good point.

Comment (jsjoeio) [reply docs/CONTRIBUTING.md]: fixed!

**Review by code-asher — APPROVED** [review]

**Pull request merged**